lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -283,16 +449,39 @@ static void populateOpConversion(RewritePatternSet &patterns,
   auto *context = patterns.getContext();
   // clang-format off
   patterns.add<
-    ConstantOpConv,
-    ConcatOpConversion,
-    ReturnOpConversion,
-    CondBranchOpConversion,
-    BranchOpConversion,
-    CallOpConversion,
-    ShlOpConversion,
-    ShrOpConversion,
-    AShrOpConversion,
-    UnrealizedConversionCastConversion
+    // Patterns of miscellaneous operations.
+    ConstantOpConv, ConcatOpConversion, ReplicateOpConversion,
+    ExtractOpConversion, ConversionOpConversion,
+
+    // Patterns of unary operations.
+    UnaryOpConversion<BoolCastOp>, UnaryOpConversion<ReduceAndOp>,
+    UnaryOpConversion<ReduceOrOp>, UnaryOpConversion<ReduceXorOp>,
```

Comment:
I think these operations might still be missing. `UnaryOpConversion` above creates a `comb::ParityOp` for all of these, which feels wrong except for `ReduceXorOp`. I think you'll want something like the following:

- `BoolCastOp`: carefully convert the operand to its "truthy" equivalent. If the operand can be converted to a simple bit vector, this should convert to that simple bit vector and then check if it is `!= 0`. All other cases we could just reject for now, since they will be more complicated (stuff like objects, classes, virtual interfaces, etc.).
- `ReduceAndOp`: should probably be something like `== <all-ones-value>`.
- `ReduceOrOp`: should probably be something like `!= 0`.
- `ReduceXorOp`: works as `ParityOp`.

Reply:
Hey, @fabianschuiki. I reimplement these conversions separately. Do I correctly understand the concept of these `UnaryOp` this time?

Reply:
So `comb.parity` just with correspondence `moore.reduceXor`. Other unary ops above are not applicable with even/odd checks. Right?